homework/list.js

```diff
@@ -1,79 +1,146 @@
 'use strict';
 
-(function() {
+(function(exports) {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
-
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
+  /**
+  * Create a instance of the ListManager.
+  *
+  * @constructor
+  * @this {ListManager}
+  */
+  var ListManager = function () {
+    /** @private */ this._listNoteContent = [];
+    /** @private */ this._wrapper = null;
   }
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+  ListManager.prototype = {
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    }
-  }
+    /**
+    * The public interface of the listManager.
+    * Fire events and do the what event describes. 
+    *
+    * @this {ListManager}
+    * @param {object} event The fired event.
+    */
+    handleEvent(event){
+      switch(event.type){
+        case 'click':
+          this.onNoteOpen(event);
+          break;
+      }
+    },
 
-  function updateList(list) {
-    _listNoteContent = list;
-  }
+    /**
+    * Initialize the listManager.
+    *
+    * @this {ListManager}
+    */
+    start() {
+      this._wrapper = document.querySelector('#note-list-wrapper');
+      this.fetchList()
+          .then(this.updateList.bind(this))
+          .then(this.drawList.bind(this))
+          .then(this.preloadFirstNote.bind(this))
+          .catch(function (error) {
```

Comment:
**Serious issue**: you should catch it, definitely. However, you need to **re-throw it** after the logging. Or the following chain will not be interrupted and may perform some unexpected effects.